components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/ConsentMgtPostAuthnHandler.java

```diff
@@ -249,6 +251,30 @@ private void removeClaimsWithoutConsent(AuthenticationContext context, ConsentCl
         removeUserClaimsFromContext(context, claimsWithoutConsent, spStandardDialect);
     }
 
+    /**
+     * Filter out the requested claims with the user attributes.
+     * @param consentClaimsData Claim metadata.
+     * @param user              Authenticated user.
+     * @return                  Filtered claim metadata.
+     */
+    private ConsentClaimsData removeClaimsNotInUserAttribute(ConsentClaimsData consentClaimsData,
+                                                             AuthenticatedUser user) {
+
+        List<ClaimMetaData> requestedClaims = consentClaimsData.getRequestedClaims();
+        List<ClaimMetaData> filteredRequestClaims = new ArrayList<>();
+        Map<ClaimMapping, String> userAttributes = user.getUserAttributes();
+        for (ClaimMetaData claimMetaData : requestedClaims) {
```

Comment:
Can `userAttributes` or `requestedClaims` be null? If so, add a null check before `for` loops.